prov/tcp/src/tcpx_ep.c

```diff
@@ -285,10 +285,25 @@ static void tcpx_ep_flush_all_queues(struct tcpx_ep *ep)
 }
 
 void tcpx_ep_disable(struct tcpx_ep *ep, int cm_err)
+{
+	struct fi_eq_err_entry err_entry = {0};
+	err_entry.err = cm_err;
+	tcpx_ep_disable_with_data(ep, &err_entry);
+}
+
+/*
+ * when sending data with UTIL_FLAG_ERROR, the util eq
+ * will store the err_entry->err_data and free it prior
+ * to the fi_eq_readerr after this data has been read.
+ *
+ * callers of tcpx_ep_disable_with_data should not
+ * free err_entry->err_data
+ */
+void tcpx_ep_disable_with_data(struct tcpx_ep *ep,
+			struct fi_eq_err_entry *err_entry)
```

Comment:
This call ends up being difficult to maintain because of unexpected behavior..  We pass in struct fi_eq_err_entry, but only expect it to be partially initialized, with the call finishing the initialization.  What's expected to be set and what's modified is not conveyed anywhere.  It would be cleaner to modify tcpx_ep_disable() directly to be:
```
void tcpx_ep_disable(struct tcpx_ep *ep, int cm_err, void *err_data, size_t size)
```
Where no err_data is available, the caller can pass in NULL,  tcpx_ep_disable() can duplication err_data as needed.